### PR TITLE
Hotfix duplicated cors headers on HTTP Request

### PR DESCRIPTION
**Pull request by Shiranuit**
## What does this PR do ?
This PR is fixes the unwanted behaviour of CORS headers being duplicated when kuzzle is responding to an HTTP Request.
Duplicating CORS headers on a Response causes the browsers that sent the request to throw a CORS error.

I also added more test cases to avoid this issue in the future.

### CODE REVIEW

**Review by Leodau — APPROVED** [review]

**Review by Aschen — APPROVED** [review]

**Review by Aschen — CHANGES_REQUESTED** [review]
Missing version bump

**Comment by Shiranuit** [comment]
**Update**: Found some code somewhere else that could duplicate headers, this is now fixed

**Added**:
- `RequestResponse.setHeaders` method now has a new parameter named `ifNotPreset`, if set to `true` only set the header if the header is not already present

File: lib/api/request/requestResponse.ts [hunk 1]
```diff
@@ -322,11 +322,16 @@ export class RequestResponse {
   /**
    * Adds new multiple headers.
    */
-  setHeaders (headers: JSONObject) {
+  setHeaders (headers: JSONObject, ifNotPresent:boolean = false) {
     assert.assertObject('headers', headers);
 
     if (headers) {
-      Object.keys(headers).forEach(name => this.setHeader(name, headers[name]));
+      Object.keys(headers).forEach(name => {
```
Comment (Aschen) [inline lib/api/request/requestResponse.ts:329]: Avoid the usage of `forEach`, prefer vanilla `for` loop

**Review by Aschen — CHANGES_REQUESTED** [review]
You must bump the package version with `npm version patch`

**Review by xbill82 — APPROVED** [review]

File: lib/api/request/requestResponse.ts [hunk 2]
```diff
@@ -322,11 +322,16 @@ export class RequestResponse {
   /**
    * Adds new multiple headers.
    */
-  setHeaders (headers: JSONObject) {
+  setHeaders (headers: JSONObject, ifNotPresent = false) {
     assert.assertObject('headers', headers);
 
     if (headers) {
-      Object.keys(headers).forEach(name => this.setHeader(name, headers[name]));
+      for (const name in headers) {
```
Comment (Aschen) [inline lib/api/request/requestResponse.ts:329]:
```suggestion
      for (const name of headers) {
```

**Review by Aschen — APPROVED** [review]

**Pull request merged**